content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+```prisma
+model Post {
+  id String @id @default(uuid())
+  commentsList Comment[]
+  tags String[]
+}
+
+type Comment {
+  text String
+}
+```
+
+##### Composite list filtering
+
+[When filtering records based on a composite list field](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types#finding-records-that-contain-composite-types-with-find-and-findmany) you will only be able to supply array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.findMany({
+  commentsList: { equals: { text: 'hello' } },
+})
+```
+
+Prisma 5:
+
+```js highlight=2-4;normal
+prisma.post.findMany({
+  commentsList: {
+    equals: [{ text: 'hello' }],
+  },
+})
+```
+
+##### Composite list input
+
+If you exclude `equals`, you still must supply an array value for composite list fields.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.findMany({
+  commentsList: { text: 'hello' },
+})
+```
+
+Prisma 5:
+
+```js highlight=2;normal
+prisma.post.findMany({
+  commentsList: [{ text: 'hello' }],
+})
+```
+
+##### Scalar lists
+
+Scalar list values must be arrays in all operations.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.create({
+  data: { tags: 'databases' },
+})
+
+prisma.post.findMany({
+  where: { tags: 'databases' },
+})
+```
+
+Prisma 5:
+
+```js highlight=2-4,8-10;normal
+prisma.post.create({
+  data: {
+    tags: ['databases'],
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: ['databases'],
+  },
+})
+```
+
+### Prisma Client embedded SQLite version updated
+
+With Prisma version 5.0.0, we have upgraded the embedded version of SQLite from `3.35.4` to `3.41.2`. We did not see any breaking changes and don't anticipate any changes needed in user projects, but if you are using SQLite, make sure to check [the SQLite changelog](https://www.sqlite.org/changes.html)
+
+### Removal of <inlinecode>jsonProtocol</inlinecode> preview feature
+
+The `jsonProtocol` preview feature is now generally available! When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol", "tracing"]
+}
+```
+
+Prisma 5:
+
+```prisma highlight=3;delete
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol"]
+}
+```
+
+In addition, several error messages have been improved. The following error message in Prisma 4 and below
+
+```text
+Argument bool: Got invalid value 'true' on prisma.createOneD. Provided String, expected Boolean.
+Argument byteA: Got invalid value 'hello prisma ⚡️🚀' on prisma.createOneD. Provided String, expected Bytes.
+Argument xml: Got invalid value 123 on prisma.createOneD. Provided Int, expected String.
+```
+
+becomes the following in Prisma 5:
+
+```text
```

Comment:
Not the best example, in my opinion, since before message is actually includes more errors in this case. Do we need to include this section? I'd maybe just mention that some of the validation error messages might look different now.

Reply:
Only the first line is the "before" equivalent to the PRisma 5 example below? So I think this could just be a normal paragraph:

> Error messages are also different, for example `foo` becomes `bar`. 

Or something like that.

Reply:
Updated! Thanks for the feedback @SevInf